src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationPath.cs

```diff
@@ -11,26 +11,44 @@ namespace Microsoft.Extensions.Configuration
     /// </summary>
     public static class ConfigurationPath
     {
+        /// <summary>
+        /// The deliminiter used for keys is ":".
+        /// </summary>
         public static readonly string KeyDelimiter = ":";
 
-        public static string Combine(params string[] pathSegements)
+        /// <summary>
+        /// Combines all several path segments into one path.
+        /// </summary>
+        /// <param name="pathSegments">The path segments to combine.</param>
+        /// <returns>The combined path.</returns>
+        public static string Combine(params string[] pathSegments)
         {
-            if (pathSegements == null)
+            if (pathSegments == null)
             {
-                throw new ArgumentNullException(nameof(pathSegements));
+                throw new ArgumentNullException(nameof(pathSegments));
             }
-            return string.Join(KeyDelimiter, pathSegements);
+            return string.Join(KeyDelimiter, pathSegments);
         }
 
-        public static string Combine(IEnumerable<string> pathSegements)
+        /// <summary>
+        /// Combines all several path segments into one path.
+        /// </summary>
+        /// <param name="pathSegments">The path segments to combine.</param>
+        /// <returns>The combined path.</returns>
+        public static string Combine(IEnumerable<string> pathSegments)
         {
-            if (pathSegements == null)
+            if (pathSegments == null)
             {
-                throw new ArgumentNullException(nameof(pathSegements));
+                throw new ArgumentNullException(nameof(pathSegments));
             }
-            return string.Join(KeyDelimiter, pathSegements);
+            return string.Join(KeyDelimiter, pathSegments);
         }
 
+        /// <summary>
+        /// Extracts the section key from a path.
```

Comment:
It is unfortunate that this so late but I just realized that I really can't understand what the API does by reading either its name or its XML docs. We happen to use it internally to obtain the value for the `Key` property for a section object but it is actually to obtain the last key for given a path, i.e. the key of the child (it doesn't matter the latter is a section or not). I recommend:

1. Rename the API to `GetChildKey()`
2. Summary: Extracts the child key part from a path.
3. Returns: Returns a string representing the last individual key found in a path.